### PR TITLE
Update histograms code, move AO metrics exporter to main package

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
This histogram code is ported from ingestion, the exporter code is just copied from the existing one.

### CODE REVIEW

File: packages/solarwinds-apm/src/appoptics/exporters/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+  export(
+    metrics: ResourceMetrics,
+    resultCallback: (result: ExportResult) => void,
+  ): void {
+    for (const scopeMetric of metrics.scopeMetrics) {
+      const scopeTags = this.#scopeTags(scopeMetric.scope)
+
+      for (const metric of scopeMetric.metrics) {
+        const name = `trace.node.${metric.descriptor.name}`
```
Comment (cheempz) [inline packages/solarwinds-apm/src/appoptics/exporters/metrics.ts:61]: This means OTel-originated metrics that get exported via apm-proto to AO will all be prefixed with `trace.node.*` (similar to runtime metrics)?

File: packages/solarwinds-apm/src/appoptics/exporters/metrics.ts [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        const temporality = metric.aggregationTemporality
+        const descriptorTags = this.#descriptorTags(metric.descriptor)
+
+        for (const dataPoint of metric.dataPoints) {
+          const dataPointTags = this.#dataPointTags(dataPoint.attributes)
+          const [tags, tagCount] = this.#oboeTags({
+            ...scopeTags,
+            ...descriptorTags,
+            ...dataPointTags,
```
Comment (cheempz) [inline packages/solarwinds-apm/src/appoptics/exporters/metrics.ts:70]: Does this mean all the above are set as the AO metric tag? Not sure what's a typical number of OTel metrics attrs to expect, but out of paranoia would it be easy to cap the max number of tags to 50 (inspired by https://github.com/solarwinds-cloud/trace/blob/master/docs/specs/custom-metrics.md) just to not rock the AO boat too much.

**Review by cheempz — COMMENTED** [review]
Now i see the very cool OTel-to-AO metrics export you mentioned! One concern is tags (cardinality) explosion for AO.

File: packages/solarwinds-apm/src/appoptics/exporters/metrics.ts (hunk 2 above)
Comment (raphael-theriault-swi) [reply packages/solarwinds-apm/src/appoptics/exporters/metrics.ts:72]: Good point, I added an upper limit.

**Review by cheempz — APPROVED** [review]
LGTM, thanks @raphael-theriault-swi!

**Pull request merged**